analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -175,8 +175,11 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) {
 
             while (parent != null) {
                 parent.scm?.let { parentScm ->
-                    parentScm.connection?.let { parentConnection ->
-                        if (parentConnection.isNotBlank() && scm.connection.startsWith(parentConnection)) {
+                    val parentScmUrl: String? = parentScm.connection ?: parentScm.url
+                    parentScmUrl?.let { url ->
+                        if (url.isNotBlank()
+                            && (scm.connection?.startsWith(url) == true || scm.url?.startsWith(url) == true)
```

Comment:
It makes me feel a bit uneasy to compare the semantically different `connection` and `url` fields here. Shouldn't we instead simply duplicate the whole check for the `url` field? Like by adding:

                    parentScm.url?.let { parentUrl ->
                        if (parentUrl.isNotBlank() && scm.url.startsWith(parentUrl)) {
                            scm = parentScm
                        }
                    }

Or maybe combine the two checks like

                    val startsWithConnection = parentScm.connection.takeUnless { it.isBlank() }
                        ?.let { scm.connection.startsWith(it) } ?: false

                    val startsWithUrl = parentScm.url.takeUnless { it.isBlank() }
                        ?.let { scm.url.startsWith(it) } ?: false

                    if (startsWithConnection || startsWithUrl) scm = parentScm